components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,240 @@
+
+        try {
+            ResourceConfig eventConfigObject = EventConfigurationMgtUtils.getEventConfig(eventKey);
+            if (eventConfigObject.getConfigs() != null &&
+                    eventConfigObject.getConfigs().containsKey(EVENT_CONFIG_SCHEMA_NAME_KEY)) {
+                return (String) eventConfigObject.getConfigs().get(EVENT_CONFIG_SCHEMA_NAME_KEY);
+            } else {
+                throw new IdentityEventServerException("Event schema not found in the resource event config " +
+                        "for the eventKey: " + eventKey);
+            }
+        } catch (EventConfigurationMgtServerException e) {
+            throw new IdentityEventServerException(e.getErrorCode(), e.getMessage());
+        } catch (ClassCastException e) {
+            throw new IdentityEventServerException(ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER.getCode(),
+                    ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER.getMessage());
+        }
+    }
+
+    /**
+     * Build the event data provider.
+     *
+     * @param event Event object.
+     * @return Event data object.
+     */
+    public static EventData buildEventDataProvider(Event event) {
+
+        Map<String, Object> properties = event.getEventProperties();
+        Map<String, Object> params = (Map)properties.get("params");
+        AuthenticationContext context = (AuthenticationContext)properties.get("context");
+        Object user = params.get("user");
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        // User information in login failed events.
+        User loginIdentifier = new User();
+        if (user instanceof AuthenticatedUser){
+            authenticatedUser = (AuthenticatedUser)user;
+            setLocalUserClaimsToAuthenticatedUser(authenticatedUser, context);
+        } else {
+            loginIdentifier = (User)user;
+        }
+
+        return EventData.builder()
+                .eventName(event.getEventName())
+                .request((HttpServletRequest)properties.get("request"))
+                .eventParams(params)
+                .authenticationContext((AuthenticationContext)properties.get("context"))
+                .authenticatorStatus((AuthenticatorStatus)properties.get("authenticationStatus"))
+                .authenticatedUser(authenticatedUser)
+                .loginIdentifier(loginIdentifier)
+                .build();
+    }
+
+    /**
+     * Retrieve the audience.
+     *
+     * @param eventUri     Event URI.
+     * @return Audience string.
+     */
+    public static SecurityEventTokenPayload buildSecurityEventToken(EventPayload eventPayload,
+                                                                    AuthenticationContext context, String eventUri)
+            throws IdentityEventException {
+
+        if (eventPayload == null) {
+            throw new IdentityEventException("Invalid event payload input: Event payload input cannot be null.");
```

Comment:
Lets standardize the errors with proper codes and messgaes

Reply:
Check https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1#discussion_r1683894493